extensions/amp-form/0.1/test/test-utils.js

```diff
@@ -24,93 +24,134 @@ describes.realWin('amp-form-utils', {}, env => {
   });
 
   describe('isFieldDefault', () => {
-    it("returns if a text field's value matches its default value", () => {
-      const html = '<input type="text" value="default">';
-      doc.body.insertAdjacentHTML('afterbegin', html);
-      const element = doc.querySelector('input');
-
-      element.value = 'not default';
-      expect(isFieldDefault(element)).to.be.false;
-
-      element.value = 'default';
-      expect(isFieldDefault(element)).to.be.true;
+    describe('text field', () => {
+      let textField;
+
+      beforeEach(() => {
+        const html = '<input type="text" value="default">';
+        doc.body.insertAdjacentHTML('afterbegin', html);
+        textField = doc.querySelector('input');
+      });
+
+      it("returns true if text field's value matches its default value", () => {
+        expect(isFieldDefault(textField)).to.be.true;
```

Comment:
You can explicitly call 
textField.value = textField.default;
in such tests, so the tests can explain themselves.

And same below.

> the test's body should be complete in that it contains all of the relevant information that a user needs to understand a test without having to track down external references and definitions.

> A test must never explicitly rely on behavior defined in setUp, or else it becomes impossible to understand the test method by itself.

> A test should specify only the relevant parameters, but also all the relevant parameters, avoiding any implicit dependency on unrelated configuration.